### PR TITLE
Archive LGPU repo

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** This PR adds the archival notice to LGPU, in preparation of of making the repo read-only at v0.32.0.

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by trbromley — COMMENTED** [review]
Thanks! Will the warning show up in the docs too?

**Comment by mlxd** [comment]
> Thanks! Will the warning show up in the docs too?

If the docs build succeeds, it will show up in latest. Will try to fix the failure, as not much info is given.

File: README.rst [hunk 1]
```diff
@@ -1,6 +1,18 @@
 PennyLane-Lightning-GPU Plugin
 ##############################
 
+.. header-start-inclusion-marker-do-not-remove
+
+❗ Warning
```
Comment (maliasadi) [inline README.rst:6]: what would you think of adding this warning to the installation page too?

I guess there should be some automated way to include this warning as a static header of all doc pages.

**Review by maliasadi — COMMENTED** [review]
Thanks @mlxd!

**Review by vincentmr — APPROVED** [review]
Looks good to me.

File: README.rst (hunk 1 above)
Comment (mlxd) [reply README.rst:6]: We've decided this is too much, and will just remove the old docs and link to the new site instead. I'll undo this.

**Review by maliasadi — APPROVED** [review]
LGTM :mammoth:

**Pull request merged**